R/zchunk_L244.building_det.R

```diff
@@ -0,0 +1,1040 @@
+    L244.StubTechIntGainOutputRatio <- L144.internal_gains %>%
+      filter(year %in% MODEL_YEARS) %>%
+      # Round and rename value
+      mutate(value = round(value, energy.DIGITS_EFFICIENCY)) %>%
+      rename(internal.gains.output.ratio = value) %>%
+      # Add region name
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      # Add building.node.input
+      left_join_error_no_match(calibrated_techs_bld_det %>%
+                                 select(supplysector, building.node.input) %>%
+                                 distinct(), by = "supplysector") %>%
+      # Add internal.gains.market.name
+      left_join_error_no_match(A44.gcam_consumer, by = "building.node.input") %>%
+      select(LEVEL2_DATA_NAMES[["TechYr"]], internal.gains.output.ratio, internal.gains.market.name)
+
+    # L244.Intgains_scalar: Scalers relating internal gain energy to increased/reduced cooling/heating demands
+    variable <- c("HDD", "CDD")
+    InternalGainsScalar_USA <- c(energy.INTERNALGAINSSCALAR_USA_H, energy.INTERNALGAINSSCALAR_USA_C)
+    US.base.scalar <- tibble(variable, InternalGainsScalar_USA)
+
+    L244.Intgains_scalar <- L244.ThermalServiceSatiation %>%
+      mutate(variable = if_else(thermal.building.service.input %in% heating_services, "HDD", "CDD")) %>%
+      left_join_error_no_match(US.base.scalar, by = "variable") %>%
+      left_join_error_no_match(L244.HDDCDD_normal_R_Y, by = c("region", "variable")) %>%
+      group_by(thermal.building.service.input) %>%
+      mutate(scalar_mult = degree.days / degree.days[region == "USA"]) %>%
+      ungroup() %>%
+      mutate(internal.gains.scalar = round(InternalGainsScalar_USA * scalar_mult, energy.DIGITS_HDDCDD)) %>%
+      select(LEVEL2_DATA_NAMES[["Intgains_scalar"]])
+
+    # Need to remove any services (supplysectors and building-service-inputs) and intgains trial markets for services that don't exist in any years
+    # L244.DeleteThermalService and L244.DeleteGenericService: Removing non-existent services, likely related to 0 HDD or CDD
+    L244.DeleteThermalService <- L244.ThermalBaseService %>%
+      group_by(region, gcam.consumer, nodeInput, building.node.input, thermal.building.service.input) %>%
+      summarise(base.service = max(base.service)) %>%
+      ungroup() %>%
+      filter(base.service == 0) %>%
+      select(-base.service) %>%
+      mutate(supplysector = thermal.building.service.input)
+
+    L244.DeleteGenericService <- L244.GenericBaseService %>%
+      group_by(region, gcam.consumer, nodeInput, building.node.input, building.service.input) %>%
+      summarise(base.service = max(base.service)) %>%
+      ungroup() %>%
+      filter(base.service == 0) %>%
+      select(-base.service) %>%
+      mutate(supplysector = building.service.input)
+
+
+
+    # ===================================================
+    # Produce outputs
+    L244.SubregionalShares %>%
+      add_title("Subregional population and income shares") %>%
```

Comment:
What are the subregions? And why are the shares all set to 1?

Reply:
This is for disaggregating the buildings sector into components that follow their own gdp and population pathways (implemented as multipliers on the whole-region pop and gdp). E.g., for urban/rural, or for South/North, without having to create new model regions. Its usage pre-dates the capability to have inter-regional trade of secondary goods, so the parameters are likely obsolete now. Still I suspect (and feel free to check me on this) that it doesn't have an assigned default value, or the default is zero, so it is required to be read in to XML.